internal/dataplane/parser/translators/ingress.go

```diff
@@ -388,3 +389,98 @@ func MaybePrependRegexPrefixForIngressV1Fn(ingress *netv1.Ingress, applyLegacyHe
 		return lo.ToPtr(MaybePrependRegexPrefix(path, regexPrefix, applyLegacyHeuristic))
 	}
 }
+
+type runeType int
+
+const (
+	runeTypeEscape runeType = iota
+	runeTypeMark
+	runeTypeDigit
+	runeTypePlain
+)
+
+func generateRewriteURIConfig(path string) (string, error) {
```

Comment:
Would you please add some comments on this function to tell about your state machine?

Reply:
I have added some brief description of the SM state and transfer process.